cpp/src/binaryop/compiled/binary_ops.cu

```diff
@@ -52,7 +54,15 @@ struct scalar_as_column_view {
       column_view(s.type(), 1, h_scalar_type_view.data(), (bitmask_type const*)s.validity_data());
     return std::pair{col_v, std::unique_ptr<column>(nullptr)};
   }
-  template <typename T, std::enable_if_t<(!is_fixed_width<T>())>* = nullptr>
+  template <typename T, std::enable_if_t<(is_struct<T>())>* = nullptr>
```

Comment:
Resolved these, there are 3 instances in this file that `enable_if` on the return type rather than the template and don't have a clear `CUDF_ENABLE_IF` replacement.

Reply:
Those can be refactored but it may be outside the scope of this PR. There are several equivalent ways to use `enable_if`. We prefer the function template parameters over conditional return types because it will simplify the refactoring towards `requires` in C++20.

https://godbolt.org/z/ob4cr99Pb

```cpp
#include <iostream>
#include <type_traits>

// With a conditional return type
template<typename T>
std::enable_if_t<std::is_same_v<T, int>, int> f(T i) { return i; }

// With SFINAE as a function template parameter
template<typename T, std::enable_if_t<std::is_same_v<T, int>>* = nullptr>
int g(T i) { return i; }

// With CUDF_ENABLE_IF
#define CUDF_ENABLE_IF(...) std::enable_if_t<(__VA_ARGS__)>* = nullptr
template<typename T, CUDF_ENABLE_IF(std::is_same_v<T, int>)>
int h(T i) { return i; }

// With a "requires" (C++20)
template<typename T> requires std::is_same_v<T, int>
int k(T i) {return i; }

int main() {
    std::cout << f(1) << std::endl;
    std::cout << g(2) << std::endl;
    std::cout << h(3) << std::endl;
    std::cout << k(4) << std::endl;
    // No matching functions for double:
    //std::cout << f(1.0) << std::endl;
    //std::cout << g(2.0) << std::endl;
    //std::cout << h(3.0) << std::endl;
    //std::cout << k(4.0) << std::endl;
}
```